cpp/autosar/src/rules/A0-1-1/UselessAssignment.ql

```diff
@@ -20,6 +20,7 @@ import codingstandards.cpp.deadcode.UselessAssignments
 
 from SsaDefinition ultimateDef, InterestingStackVariable v
 where
+  not isExcluded(ultimateDef, DeadCodePackage::uselessAssignmentQuery()) and
   not isExcluded(v, DeadCodePackage::uselessAssignmentQuery()) and
```

Comment:
```suggestion
  not isExcluded(v, DeadCodePackage::uselessAssignmentQuery()) and
```

Let remove the exclusion on the variable itself to prevent confusion around the cases

```
int x; // suppress all useless assignments of x

x = 0;

if (...) {
  x = 1;
} else {
  x = 2;
  x = 3;
}
```

and

```
int x = 0; // suppress all or only the current useless assignement

if (...) {
  x = 1;
} else {
  x = 2;
  x = 3;
}
```